tests/lib/rules/prefer-const.js

```diff
@@ -133,6 +153,16 @@ ruleTester.run("prefer-const", rule, {
             code: "(function() { let x; { x = 0; foo(x); } })();",
             parserOptions: { ecmaVersion: 6 },
             errors: [{ message: "'x' is never modified, use 'const' instead.", type: "Identifier"}]
+        },
+        {
+            code: "let x; for (const a of [1,2,3]) { x = foo(); bar(x); }",
+            parserOptions: { ecmaVersion: 6 },
```

Comment:
I'm not sure this is correct. `x = foo()` could return a different value each time it's called. I don't think you can safely replace this with a const.